src/operator/operator_tune-inl.h

```diff
@@ -356,7 +356,9 @@ class OperatorTune : public OperatorTuneByType<DType> {
   static duration_t GetOMPLoopOverhead() {
     // It was found empirically that OMP times was not heavily tied to number of cores,
     // so take an average across all core counts
-    const auto max_cores = static_cast<size_t>(omp_get_num_procs()) >> 1;
+    const auto max_cores_default = static_cast<size_t>(omp_get_num_procs()) >> 1;
```

Comment:
That's just diving it by two, but according to your statement and the comment, this process does not benefit of many codes. Should we maybe instead use max (4, num_cores) as default?

Otherwise, the default behaviour would not change and users would still face the same challenges. I'm not a big fan of having to know a "secret" environment variable. Of course, we can have this variable to offer users the possibility to set it specifically, but generally we should choose a good default

Reply:
"Should we maybe instead use min (3, num_cores) as default" 
One way to make this process fast was reducing num of iterations it was taking to determine if OMP should be used or not.
The reduction in outside loop count reduces the overhead of tuning code significantly. 
I tested specifically with 72 cores and and it took around 40 sec to launch 30 mxnet processes which is much better than existing behavior.
I am not sure about if we use 3 cores, that would be right behavior because OMP badness can increase with increase in num_cores. So tuning code can say OMP is good as it was tested on 3 cores, but the real code may behave differently.

Reply:
I see. To clarify, these 3 cores are just for the tuning stage, right?

@cjolivier01 @TaoLv @pengzhao-intel could you maybe provide some guidance here? I think you guys already made a lot of experience with open mp

Reply:
"these 3 cores are just for the tuning stage" - that is correct.